### ISSUE TITLE
[BUG] reference to old numpy.trapz causes ClaSPTransformer to break with default parameters

### ISSUE THREAD

**Issue by pygabr** [alan-turing-institute/sktime#10759]
Hi, the ClaSPTransformer uses the ROC_AUC as default scoring metric. This appears to be calculated by using the numpy.trapz function, which however has been deprecated and removed from numpy, replaced with trapezoid. 
This causes the transform method of the transformer to break down.

I note that with sktime 1.1.0 the numpy version is constrained between >=1.21 and <2.5, however I understand that all numpy versions >2.0 would trigger the bug described.

**To Reproduce**
import numpy
from sktime.transformations.clasp import ClaSPTransformer

a=numpy.arange(100)
c=ClaSPTransformer()
c.fit_transform(a)

Thank you

**Comment by fkiraly** [comment]
@pygabr, this is already fixed in https://github.com/sktime/sktime/pull/10727, but has not been moved to release yet. The next release should fix the issue.

**Comment by fkiraly** [comment]
> however I understand that all numpy versions >2.0 would trigger the bug described.

They would trigger a warning only. This was a deprecation warning we missed, from 2.5 onwards it causes an exception.

**Comment by fkiraly** [comment]
would be great if you could check if your issue is resolved on the current development branch (`main`)

**Comment by pygabr** [comment]
> would be great if you could check if your issue is resolved on the current development branch (`main`)

apart from a smart app control-triggered block of numba (which I assume is completely unrelated), yes it works, thank you. 
Apologies I missed #10727
